planetscale/databases.go

```diff
@@ -36,13 +36,22 @@ type DeleteDatabaseRequest struct {
 	Database     string
 }
 
+// PromoteBranchRequest encapsulates the request for promoting a branch to
+// production.
+type PromoteBranchRequest struct {
+	Organization string `json:"-"`
+	Database     string `json:"-"`
+	Branch       string `json:"branch"`
+}
+
 // DatabaseService is an interface for communicating with the PlanetScale
 // Databases API endpoint.
 type DatabasesService interface {
 	Create(context.Context, *CreateDatabaseRequest) (*Database, error)
 	Get(context.Context, *GetDatabaseRequest) (*Database, error)
 	List(context.Context, *ListDatabasesRequest) ([]*Database, error)
 	Delete(context.Context, *DeleteDatabaseRequest) error
+	PromoteBranch(context.Context, *PromoteBranchRequest) (*DatabaseBranch, error)
```

Comment:
I wonder if this method should be part of `DatabaseBranchesService` interface because it's branch related? It's also the only method here that returns a different type.

Reply:
I had asked this question on Slack as well, but we decided that given that the URL structure is `/databases/:db/promote-branch`, we kept it in that interface.